### ISSUE TITLE
Allow toggle to toggle stroke commands like +hook

### ISSUE THREAD

**Issue by Scrumplex** [teeworlds/teeworlds#2697]
I am currently tinkering around with my teeworlds configuration and I would really like to have a way to toggle a stroke command like `+hook` or `+fire` (anything with a `+`) with the `toggle` command.

Here is what I am trying to do:

```
bind m "toggle +fire"
```
Which of course tells me that I need two more parameters.

Settings those parameters to anything didn't work.

So I would suggest either adding another command like `toggle_stroke` or something like that that can handle stroke commands, or by just expanding the function of the currently available `toggle` command. For any contributions: here is the command handler of `toggle`: https://github.com/teeworlds/teeworlds/blob/16b5c961bfe073ceab27978186a6cbbf386225b6/src/engine/shared/console.cpp#L673

**Comment by Dune-jr** [comment]
I don't understand what `toggle +fire` is supposed to do differently than just `+fire`. Can you explain?

**Comment by Scrumplex** [comment]
> what `toggle +fire` is supposed to do differently than just `+fire`

It should allow me to just press the key once and then let `+fire` be active until I press my key again. Of course in the case of fire it doesn't make much sense. But it's just an example

**Comment by Dune-jr** [comment]
`+fire` is not a flag though, it is an event.
For anything that is a flag, for example `cl_dynamic_camera`, it works: `bind m "toggle cl_dynamic_camera 1 0`

**Comment by Scrumplex** [comment]
Yes. And that's what I am requesting here: Make it work with an event (or however it is called)

**Comment by Patiga** [comment]
Regardless of how it works technically, i think that it opens up movement options not yet possible. i.e. moving a fraction of a block to the side by binding toggle +left on 2 keys, pressing them right after each other (this is what i came up with quickly, but i believe much more could be possible). In my opinion, this shouldn't be added to the client.

**Comment by Learath2** [comment]
It wouldn't make much sense with `+fire`, but I guess something like `+hook` would make more sense. Like toggling between holding the stroke down or releasing it.

**Comment by Dune-jr** [comment]
Oh you're right @Learath2 
I'm not sure we'd want this sort of stuff in the default client either. Worth noting, there was a 0.5 bug that would let you do such things

**Comment by Scrumplex** [comment]
So this would be something that should be implemented downstream instead? I actually want to use this for one of the events added by DDNet. I just thought it would be appropriate to build this into upstream.

**Comment by Dune-jr** [comment]
It's not unreasonable to open up a disucussion about it

**Comment by SoulyVEVO** [comment]
it should be possible using ingame `exec` command
